_P201_VBAN.ino

```diff
@@ -0,0 +1,577 @@
+
+      
+      adc1_config_channel_atten(ADC1_CHANNEL_1, ADC_ATTENUATION); //ADC_ATTEN_11db); // connects the ADC 1 with channel 7 (GPIO 35)
+      adc1_get_raw(ADC1_CHANNEL_1); //we need this to initialize the ADC correctly
+      }
+
+      xTaskCreate(adcHandler, "Handler Task", 5000, NULL, 1, &adcHandlerTask);
+
+      //timer = timerBegin(0, 80, true); // 80 Prescaler => 1 microseconds for 8Khz
+      timer = timerBegin(0, 40, true); // 40 Prescaler => 0.5 microseconds  for 16Khz
+      //timer = timerBegin(0, 20, true); // 20 Prescaler => 0.25 microseconds  for 32Khz
+      timerAttachInterrupt(timer, &onTimer, true); // binds the handling function to our timer 
+      timerAlarmWrite(timer, 125, true); //x*8Khz sample rate
+      timerAlarmEnable(timer);
+
+
+      //after the plugin has been initialised successfuly, set success and break
+      log = F("VBAN: Initialized");
+      addLog(LOG_LEVEL_INFO,log);
+      success = true;
+      break;
+
+    }
+
+    case PLUGIN_READ:
+    {
+      //code to be executed to read data
+      //It is executed according to the delay configured on the device configuration page, only once
+
+      //after the plugin has read data successfuly, set success and break
+      success = true;
+      break;
+
+    }
+
+    case PLUGIN_WRITE:
+    {
+      //this case defines code to be executed when the plugin executes an action (command).
+      //Commands can be accessed via rules or via http.
+      //As an example, http://192.168.1.12//control?cmd=dothis
+      //implies that there exists the comamnd "dothis"
+
+      //if (plugin_not_initialised)
+       // break;
+
+      //implement any commands here
+
+       break;
+    }
+
+	case PLUGIN_EXIT:
+	{
+	  //perform cleanup tasks here. For example, free memory
+    free(buffer);
+    buffer= NULL;
+	  break;
+
+	}
+    
+    case PLUGIN_ONCE_A_SECOND:
+    {
+      //code to be executed once a second. Tasks which do not require fast response can be added here
+      UserVar[event->BaseVarIndex] = audio_power;
+      success = true;
+      
+      break;
+    }
+  
+    /*
+    case PLUGIN_TEN_PER_SECOND:
+    {
+      //code to be executed 10 times per second. Tasks which require fast response can be added here
+      //be careful on what is added here. Heavy processing will result in slowing the module down!
+      
+      success = true;
+      break;
+    
+    
+    }
+
+  
+    case PLUGIN_FIFTY_PER_SECOND:
+    {
+
+      success = true;
+      break;
+    }
+    */
+  }   // switch
+  return success;
+
+}     //main plugin function
+
+
+
+void adcHandler(void *param) {
+  while (true) {
+    // Sleep until the ISR gives us something to do, or for 1 second
+    ulTaskNotifyTake(pdFALSE, pdMS_TO_TICKS(1000));  
+    if (WiFiConnected()) 
+    if (transmitNow>=0){
+        uint16_t * audio = (uint16_t *)(buffer+transmitNow);
+
+        P201_setFrameCounter((buffer+transmitNow),frameCounter);

```

Comment:
You just created a pointer before calling this function, why not use `audio` as parameter?

Reply:
because of the pointer size.
the pointer audio is used for any post processing and calculation to access the 12bit audio 
The frameCounter and UDP require the pointer to bytes.

Reply:
Ah I get it 👍